opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -253,6 +263,11 @@ public long timestampMicros() {
             this.context = new MockContext(parent.traceId, nextId(), parent.baggage);
             this.parentId = parent.spanId;
         }
+
+        if (tracer.getActiveSpanManager() != null) {
```

Comment:
would need to do something to get around all of these checks. I think it's going to be racy to have the ASM settable at something other than construction-time for a Tracer.